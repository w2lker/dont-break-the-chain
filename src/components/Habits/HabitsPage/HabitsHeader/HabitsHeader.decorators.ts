import withStyles from '@material-ui/core/styles/withStyles';
import HabitsHeader from './HabitsHeader';
import habitsHeaderStyles from './HabitsHeader.styles';

const styledComponent = withStyles(habitsHeaderStyles)(HabitsHeader);

export default styledComponent;
