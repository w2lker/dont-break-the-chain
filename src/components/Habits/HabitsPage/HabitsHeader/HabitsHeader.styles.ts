import { Styles } from '@material-ui/styles/withStyles';
import { bon } from '../../../../utils/styleUtils';

const habitsHeaderStyles: Styles<any, any> = {
  root: {
    display: 'block',
    textAlign: 'center',
    boxShadow: '0px 4px 8px rgba(176, 190, 197, 0.24)',
    marginBottom: 5,
  },
  title: {
    ...bon.setFont(20, '#263238', 500, 0.15),
    margin: 0,
    padding: '15px 0',
    lineHeight: '28px',
  },
};

export default habitsHeaderStyles;
