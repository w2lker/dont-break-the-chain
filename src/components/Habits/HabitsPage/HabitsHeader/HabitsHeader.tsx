import React from 'react';

import { WithStyles } from '@material-ui/styles';

import defaultTexts from '../../../../contants/defaultTexts';
import habitsHeaderStyles from './HabitsHeader.styles';

export interface IHabitsHeaderProps extends WithStyles<typeof habitsHeaderStyles> {}

const HabitsHeader: React.FC<IHabitsHeaderProps> = (props) => {
  const { classes } = props;
  const texts = defaultTexts.habits.header;
  return (
    <div className={classes.root}>
      <h4 className={classes.title}>{texts.title}</h4>
    </div>
  );
};

export default HabitsHeader;
