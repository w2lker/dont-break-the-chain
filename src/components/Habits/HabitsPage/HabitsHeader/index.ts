import HabitsHeader from './HabitsHeader.decorators';

export default HabitsHeader;
