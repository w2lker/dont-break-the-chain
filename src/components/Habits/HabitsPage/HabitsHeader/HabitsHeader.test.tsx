import React from 'react';

import { mount, shallow } from 'enzyme';
// import cases from 'jest-in-case';

import HabitsHeader, { IHabitsHeaderProps } from './HabitsHeader';
import HabitsHeaderDecorated from './HabitsHeader.decorators';
import habitsHeaderStyles from './HabitsHeader.styles';

const testSetup = () => {
  const classes = {
    root: 'habitsHeader-root',
    title: 'habitsHeader-title',
  };
  const classesKeys = Object.keys(classes);
  const sampleProps: IHabitsHeaderProps = {
    classes,
  };
  return { classes, classesKeys, sampleProps };
};

describe('HabitsHeader component', () => {

  it('match snapshot', () => {
    const { sampleProps } = testSetup();
    const component = shallow(<HabitsHeader {...sampleProps} />);
    expect(component.debug()).toMatchSnapshot();
  });

});

describe('HabitsHeader styles', () => {
  it('contains principle classes', () => {
    const { classesKeys } = testSetup();
    expect(habitsHeaderStyles).toBeDefined();
    classesKeys.forEach((keyValue) => {
      // @ts-ignore
      expect(habitsHeaderStyles[keyValue]).toBeDefined();
    });
  });
});

describe('HabitsHeader decorators', () => {
  it('provides styled classes from decorators', () => {
    const { classesKeys } = testSetup();
    const component = mount(<HabitsHeaderDecorated />);
    // @ts-ignore
    const assignedClasses = component.find('HabitsHeader').props().classes;
    classesKeys.forEach(keyValue => {
      expect(assignedClasses[keyValue]).toBeDefined();
    });
  });
});
