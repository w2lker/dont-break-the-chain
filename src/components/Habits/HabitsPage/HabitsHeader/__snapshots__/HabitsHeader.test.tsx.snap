// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitsHeader component match snapshot 1`] = `
"<div className=\\"habitsHeader-root\\">
  <h4 className=\\"habitsHeader-title\\">
    My Habits
  </h4>
</div>"
`;
